### ISSUE TITLE
Results on Person ReID

### ISSUE THREAD

**Issue by Xiangyu-CAS** [Andrew-Brown1/Smooth_AP#1]
Hi, I was trying to apply SmoothAP into Person ReID but didn't get desirable result. Face retrieval experiments are in your paper but details are still not clear.  So I am not sure which part I did wrong, could you help me ?

Setting.
MODEL: resnet50 + BN1d + L2Norm
Loss: cosface + smoothAP
Batch: batch-size=192, pk sampler, p=16 (p=16 outperforms p=4 with a large margin) 

My Experiments.
|method|mAP|
|--------|------|
|cosface|36.9%|
|cosface+triplet|37.5%|
|cosface+smoothAP|34.6%|
|smoothAP|29.9%|

**Comment by Andrew-Brown1** [comment]
Hi,

Thank you for your interest. I plan on releasing some code for training when I am back from holiday in 2 weeks. This might be of help. 

Does your pk sampler ensure that all samples classes are different in a batch? 
The version of the Smooth-AP loss that I uploaded can only work if this is the case. Also what value for tau are you using?

**Comment by Xiangyu-CAS** [comment]
Thank you for quick reply.
1. Does "all samples classes are different in a batch" mean p=1? I printed labels which listed below
2. what is tau refer to ?


```
tensor([641, 641, 641, 641, 641, 641, 641, 641, 641, 641, 641, 641, 641, 641,
        641, 641, 140, 140, 140, 140, 140, 140, 140, 140, 140, 140, 140, 140,
        140, 140, 140, 140, 502, 502, 502, 502, 502, 502, 502, 502, 502, 502,
        502, 502, 502, 502, 502, 502, 181, 181, 181, 181, 181, 181, 181, 181,
        181, 181, 181, 181, 181, 181, 181, 181,  69,  69,  69,  69,  69,  69,
         69,  69,  69,  69,  69,  69,  69,  69,  69,  69,  82,  82,  82,  82,
         82,  82,  82,  82,  82,  82,  82,  82,  82,  82,  82,  82, 120, 120,
        120, 120, 120, 120, 120, 120, 120, 120, 120, 120, 120, 120, 120, 120,
        200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200,
        200, 200, 509, 509, 509, 509, 509, 509, 509, 509, 509, 509, 509, 509,
        509, 509, 509, 509, 451, 451, 451, 451, 451, 451, 451, 451, 451, 451,
        451, 451, 451, 451, 451, 451, 617, 617, 617, 617, 617, 617, 617, 617,
        617, 617, 617, 617, 617, 617, 617, 617, 133, 133, 133, 133, 133, 133,
        133, 133, 133, 133, 133, 133, 133, 133, 133, 133], device='cuda:0')

```

**Comment by WeidiXie** [comment]
Please check Section 4.1 in the paper.

**Comment by Andrew-Brown1** [comment]
So if p = samples per class, k = number of classes. It is essential that k ‘different’ classes are sampled in every batch. The batch labels you sent below look fine, but make sure that this rule is asserted. Some pk samplers do not assert this.

Tau is called ‘anneal’ in the loss function initialisation :)

**Comment by Xiangyu-CAS** [comment]
@Andrew-Brown1 @WeidiXie Thank you, I found it. tau is setted to 0.01 in my experiments. I also tried tau=0.1 and 0001, but which turned out inferior to 0.01

**Comment by WeidiXie** [comment]
The training code will be released in two-three weeks time, stay tuned, thanks.

**Comment by Xiangyu-CAS** [comment]
@WeidiXie Oh good, looking forward to it. : ) 
BTW, do you have a plan to combine it with an memory bank, since batch size really matters?

**Comment by Andrew-Brown1** [comment]
Thanks!
So perhaps in the future. Many previous works have explored it and showed it to be beneficial.

**Comment by XuShoweR** [comment]
> The training code will be released in two-three weeks time, stay tuned, thanks.

Looking forward to your training code!
